tempodb/backend/local/compactor.go

```diff
@@ -20,14 +21,37 @@ func (rw *Backend) MarkBlockCompacted(blockID uuid.UUID, tenantID string) error
 
 func (rw *Backend) ClearBlock(blockID uuid.UUID, tenantID string) error {
 	if len(tenantID) == 0 {
-		return fmt.Errorf("empty tenant id")
+		return errors.New("empty tenant id")
 	}
 
 	if blockID == uuid.Nil {
-		return fmt.Errorf("empty block id")
+		return errors.New("empty block id")
 	}
 
-	return os.RemoveAll(rw.rootPath(backend.KeyPathForBlock(blockID, tenantID)))
+	path := rw.rootPath(backend.KeyPathForBlock(blockID, tenantID))
+	err := os.RemoveAll(path)
+	if err != nil {
+		return fmt.Errorf("failed to remove keypath for block %s: %w", path, err)
+	}
+
+	// if there are no more blocks in this tenant, clear the tenant
+	path = rw.rootPath(backend.KeyPath{tenantID})
+	folders, err := os.ReadDir(path)
+	if errors.Is(err, os.ErrNotExist) {
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("failed to list keypath for tenant %s: %w", path, err)
+	}
+
+	if len(folders) == 0 {
+		err = os.RemoveAll(path)
```

Comment:
There shouldn't be a race condition here between blocks because this is always done under mutex lock correct?

Reply:
I actually don't know. Generally I've considered the backends concurrency safe b/c they are protected by the file system or object storage's own ordering of events. I think this is a point in favor of removing it.